stacks-signer/src/v0/signer.rs

```diff
@@ -538,15 +538,7 @@ impl Signer {
             .signer_db
             .block_lookup(self.reward_cycle, &signer_signature_hash)
         {
-            Ok(Some(block_info)) => {
-                if block_info.state == BlockState::GloballyRejected
-                    || block_info.state == BlockState::GloballyAccepted
-                {
-                    debug!("{self}: Received block validation for a block that is already marked as {}. Ignoring...", block_info.state);
-                    return None;
-                }
-                block_info
-            }
+            Ok(Some(block_info)) => block_info,
```

Comment:
Should we check if the state is locally accepted and not broadcast, since that would indicate we have already broadcast our response. I think that would only happen if another signer attached to the same stacks-node requested validation for the same block after we've already approved it.

Reply:
hm... but that would indicate that this second signer got the block proposal way after this signer, and i'm not sure how that would happen if they're both event observers?

Reply:
Yeah, I think it would definitely be weird for it to happen, but I don't think it hurts to check.

Reply:
Oh wait, could that case also happen if the miner re-proposes the same block? So maybe we should go back to what you had, and always send the response. The miner will just ignore it if it is a duplicate. Sorry for that detour.

Reply:
I think at least one signer runs multiple signer binaries attached to the same node, so it could happen.

Reply:
> Oh wait, could that case also happen if the miner re-proposes the same block? So maybe we should go back to what you had, and always send the response. The miner will just ignore it if it is a duplicate. Sorry for that detour.

Discussed further, and we think this is good as-is.